enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -307,6 +311,26 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
 
     }
 
+    private void checkTokenEnv(JWTClaimsSet claims, String gatewayEnv) throws APISecurityException {
+        // If the claim "aud" does not exist, getAudience() returns an empty list.
+        // If the value for "aud" is a String, getAudience() appends the value to the list that is returned.
+        List<String> aud = claims.getAudience();
+        for (String item: aud) {
+            if (item.startsWith(APIConstants.JwtTokenConstants.ENV_NAME_PREFIX)) {
+                String choreoGatewayEnv = APIConstants.JwtTokenConstants.ENV_NAME_PREFIX + gatewayEnv;
+                if (aud.contains(choreoGatewayEnv)) {
+                    log.debug("Environment validation for the access token was successful.");
+                } else {
+                    log.debug("The access token does not have access to the environment {}.", gatewayEnv);
+                    throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                            APISecurityConstants.API_AUTH_INVALID_ENVIRONMENT,
+                            APISecurityConstants.API_AUTH_INVALID_ENVIRONMENT_ERROR_MESSAGE);
```

Comment:
We get to throw the exception because of the `aud.contains(choreoGatewayEnv)` method. Since aud is an array, we get to directly check if "choreo:environment:<env_name>" is there in the list, which may include other envs as well. We are using a kind of a reverse logic here.